cs/src/Management/TunnelUserAgent.cs

```diff
@@ -47,4 +49,40 @@ public static class TunnelUserAgent
 
         return new ProductInfoHeaderValue(productName, productVersion ?? "unknown");
     }
+
+    /// <summary>
+    /// Gets machine properties and adds them to the user agent.
+    /// Properties include os, os version, and windows
+    /// partner id if applicable
+    /// </summary>
+    /// <returns>List of product info header values with
+    /// machine properties.</returns>
+    public static List<ProductInfoHeaderValue> GetMachineHeaders()
+    {
+        var windowsHeaderValues = new List<ProductInfoHeaderValue>();
+        var os = Environment.OSVersion.Platform;
+        windowsHeaderValues.Add(ProductInfoHeaderValue.Parse("OS" + "/" + os.ToString()));
+        var version = Environment.OSVersion.Version;
```

Comment:
Use `RuntimeInformation.OSDescription` to get the OS name and version; I think it will be more detailed than `Environment.OSVersion` for non-Windows platforms.